tests/lib/rules/sort-keys.js

```diff
@@ -506,6 +506,50 @@ ruleTester.run('sort-keys', rule, {
     {
       code: 'var obj = {a:1, _:2, b:3}',
       options: ['desc', { natural: true, caseSensitive: false, minKeys: 4 }]
+    },
+
+    // runOutsideVue (false) should ignore unsorted keys outside of vue
+    {
+      code: 'var obj = { c:3, a:1, b:2}',
+      options: ['asc', { runOutsideVue: false }]
+    },
+    {
+      filename: 'test.js',
+      code: `
+        const { component } = Vue;
+        component('test', {
+          name: 'app',
+          data () {
+            return {
+              c: 3,
+              a: 1,
+              b: 2
+            }
```

Comment:
Thank you for this PR!
Why is this not reported? I think `a`, `b`, and `c` are Vue component properties and should be sorted.

Reply:
Thank you for the review.

I have pushed a change that will ensure objects returned by the `data()` method are sorted.

Just clarifying here, we only want to sort the keys of `data()` objects and we can leave objects defined within `computed`, `watchers` and `methods` unsorted?

If that is the case then this is all good.

If we want to also sort all new objects defined within a vue instance but not directly attached to the component maybe we should look towards implementing a new option that only sorts keys that are directly attached to the vue instance.

For example
```
...
computed: {
  // I would like the computed keys to be sorted.
  a () {
    // But I do not want to sort new objects defined within my computed.
     return {
       b: null,
       a: null,
     }
  },
}
```

Reply:
Sorry for late reply.

I think this rule should sort the keys of all objects that aren't reserved as Vue component definitions.
Therefore, I think the following `b` and `a` should also be sorted. Because they are *inside* the Vue component.

```vue
computed: {
  a () {
     return {
       b: null,
       a: null,
     }
  },
}
```

If you want to sort only the properties of Vue components (e.g. props, data, computed, watch, and methods),  I think we can add another rule to exclude it from being checked by the `vue/sort-keys` rule.
I think it would be better to have a separate rule for sorting related to Vue components than `vue/sort-keys` rule,  like `vue/order-in-components` rule.
https://eslint.vuejs.org/rules/order-in-components.html
What do you think?

Reply:
Hello, sorry for the slow reply, been very busy with work lately.

This rule already accepts an `ignoreGrandchildrenOf` option that might be applicable here, iirc it only considers grandchildren keys of the same object, not newly created objects. Let me confirm this behaviour when I get a chance and maybe we could look towards making that ignore all grandchildren, not just grandchildren that are directly attached to the grandparent key.

Reply:
The ignoreGrandchildrenOf option is an option to avoid conflicting with ordering conventions with Vue components.    
If we need an option to ignore all grandchildren, I don't think that option should ignore the outside of the Vue component.  

I think `vue/sort-keys` rule should have a role to enforce an order other than the Vue component ordering convention.  
I think the order of Vue properties is the ordering convention of Vue components, so I think we need another rule.

I think `runOutsideVue=false` is just an option to ignore the outside of Vue. (However, I'm not sure if it's necessary for the user.)

I think it should be another rule if you want to enforce the order of Vue properties.
Where do you want to enforce the order?

Reply:
@ota-meshi I was hoping to enforce order on all `props`, `computed` props, `methods` and `watch` child keys, eg.

```
export default {
  // These keys would not be enforced by this rule/option combination.
  name: 'ComponentName',

  props: {
    // These keys order would be enforced by this rule/option combination
    a: { type: String, default: null },
    b: { type: String, default: null },
  },
  
  computed: {
    c () {
      return {
        // These keys would not be enforced by this rule/option combination.
        e: null,
        d: null,
      }
    }
  },
}
```
I agree, I believe a new rule would be more appropriate for this. I have closed this PR for now and if I get a chance in the future I will submit a new PR, thanks.